qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -126,17 +136,17 @@ def run(self, dag):
                         basis_count += 1
                     subcirc.append(nd.op, [q[block_index_map[i]] for i in nd.qargs])
                 unitary = UnitaryGate(Operator(subcirc))  # simulates the circuit
-                if self.force_consolidate or unitary.num_qubits > 2 or \
-                        self.decomposer.num_basis_gates(unitary) < basis_count:
 
-                    if len(block_qargs) <= 2:
-                        new_dag.apply_operation_back(
-                            self.decomposer(unitary).to_gate(),
-                            sorted(block_qargs, key=lambda x: block_index_map[x]))
-                    else:
-                        new_dag.apply_operation_back(
-                            UnitaryGate(unitary),
-                            sorted(block_qargs, key=lambda x: block_index_map[x]))
+                max_2q_depth = 20  # If depth > 20, there will be 1q gates to consolidate.
```

Comment:
why is this needed?

Reply:
This is a stopgap to ensure we trigger synthesis in cases where we know there can be a reduction in 1q gate depth (since there's no way to gauge reductions in 1q gate depth without synthesizing the gate). This can be removed with #3658 .